contracts/AstarBase.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.7;
+
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./DappsStaking.sol";
+import "./SR25519.sol";
+
+/// @author The Astar Network Team
+/// @title Astarbase. Mapping of Stakers ss58 <> H160
+contract AstarBase is Ownable {
+    using Counters for Counters.Counter;
+    Counters.Counter public registeredCnt;
+    bytes PREFIX= hex"3c42797465733e";
+    bytes POSTFIX= hex"3c2f42797465733e";
+    bool public paused;
+    mapping(address => bytes32) public addressMap;
+    DappsStaking public constant DAPPS_STAKING = DappsStaking(0x0000000000000000000000000000000000005001);
+    SR25519 public constant SR25519Contract = SR25519(0x0000000000000000000000000000000000005002);
+
+    constructor() {
+        paused = false;
+    }
+
+    /// @notice Register senders' address with corresponding SS58 address and store to mapping
+    /// @param polkadotPublicKey, SS58 address used for staking
+    /// @param signedMsg, signed message
+    /// @param message, message string that was signed
+    function register(bytes32 polkadotPublicKey, bytes calldata signedMsg, string calldata message) external {
```

Comment:
Why is the message part of the param? I thought we have a string const within the contract and let this function create the message.

For example:
```
const MSG_PREFIX = "sign this to register to AstarBase for:";

function register(bytes32 polkadotPublicKey, bytes calldata signedMsg) {
// ..

var message = MSG_PREFIX + polkadotPublicKey;

verifyPolkadot(polkadotPublicKey, signedMsg, fullMessage);

}
```

something like this.

In other words, the message should be constructed from the user address instead of making the user pass the same data twice. I hope this makes sense

Reply:
We could do it like that ofc, but since frontend needs to initiate signing with SS58 account and needs to send it to smart contract for the verification/registration, imo it makes more sense to have this string defined in the front-end. It also gives more flexibility to the smart contract. You may also want to have different message on Astar/Shiden

Reply:
Generally, it is a good security practice to let the backend decide the message so that the contract won't be hacked even if the front end is compromised. I know this contract doesn't have any security-critical logic that affects the chain, but why don't we use best practices?

From what I can see, this function only checks if the provided address and the signature match the provided message. With the current logic, it is possible to register an address without actually having to own that address.
I suggest that the message be constructed within the contract in the combination of the H160 address and the SS58 address string.

Reply:
changed as suggested